examples/performance-debugging/performance-debugging.cpp

```diff
@@ -324,7 +324,8 @@ void print_usage(const char *filename)
 }
 
 
-void print_vector(const gko::matrix::Dense<> *vec)
+template <typename ValueType>
```

Comment:
The only question left here concern the three functions `get_norm` and related on top of this file. We could make them return a `remove_complex` of `ValueType` instead of plain `double`, but if we don't want to complicate this too much as is should be fine.

Reply:
I kind of thought `ValueType` would only be `float` or `double`, but you are correct, if we also want to support `std::complex<float>` and `std::complex<double>`, we would also need the `remove_complex` for the type of the `reduction_factor`.

Reply:
Yes that's true, whenever we think of complex valuetypes, then we need remove complex for norms and reduction factors. Another thing to look out for is the matrix initialization sometimes. 

In general, Ginkgo should support complex computations (definitely reference kernels do). Of course not all of CUDA supports complex, but since we are going this direction ensuring that it works for these would be great. I'm fairly certain some users will try out complex in the examples at some point, so it's better if we tried it ourselves first.

Reply:
Then we need to test every single example for compatibility with `std::complex<double>` since I am fairly certain we are missing some `gko::remove_complex<>` in the current examples code.
I think for now, checking if it compiles should be enough (to detect the missing `remove_complex` parts), but at some point, we need to test our whole library for compatibility with complex numbers.

Reply:
Yes, this makes sense. The problem is it might make some examples verbose. Maybe I can add it only for a few examples ? Or should I add it for all ?